sympy/series/limits.py

```diff
@@ -136,6 +136,11 @@ def doit(self, **hints):
         if not e.has(z):
             return e
 
+        # limit can work with binomial by converting to factorial first
+        if (e.has(binomial) and not (hints.get('sequence', True)
```

Comment:
Sorry, I didn't see the patch you applied, correctly. 
Yes i'm getting the same errors if we implement `_eval_rewrite_as_tractable` to return gamma (oo) and to return factorial (doesn't evaluate) . 
So what should be done?